crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -0,0 +1,181 @@
+    /// The radius of the circle
+    pub radius: f32,
+}
+impl Primitive2d for Circle {}
+
+/// An unbounded plane in 2D space. It forms a separating surface trough the origin,
+/// stretching infinitely far
+#[derive(Clone, Copy, Debug)]
+pub struct Plane2d {
+    /// The normal of the plane, the plane will be placed perpendicular to this direction
+    pub normal: Direction2d,
+}
+impl Primitive2d for Plane2d {}
+
+/// An infinite line along a direction in 2D space.
+///
+/// For a finite line: [`Segment2d`]
+#[derive(Clone, Copy, Debug)]
+pub struct Line2d {
+    /// The direction of the line, the line extends infinitely in both the positive
+    /// and negative of this direction
+    pub direction: Direction2d,
+}
+impl Primitive2d for Line2d {}
+
+/// A segment of a line along a direction in 2D space.
+#[doc(alias = "LineSegment2d")]
+#[derive(Clone, Debug)]
+pub struct Segment2d {
+    /// The direction of the line
+    pub direction: Direction2d,
+    /// Half the length of the line segment, the segment extends by this amount in both
+    /// the positive and negative direction
+    pub half_length: f32,
+}
+impl Primitive2d for Segment2d {}
+
+impl Segment2d {
+    /// Create a line segment from a direction and full length of the line
+    pub fn new(direction: Direction2d, length: f32) -> Self {
+        Self {
+            direction,
+            half_length: length / 2.,
+        }
+    }
+
+    /// Get a line segment and translation from two points at each end of a line segment
+    ///
+    /// Panics if point1 == point2
+    pub fn from_points(point1: Vec2, point2: Vec2) -> (Self, Vec2) {
+        let diff = point2 - point1;
+        let length = diff.length();
+        (
+            Self::new(Direction2d::from_normalized(diff / length), length),
+            (point1 + point2) / 2.,
+        )
+    }
+
+    /// Get the position of the first point on the line segment
+    pub fn point1(&self) -> Vec2 {
+        *self.direction * -self.half_length
+    }
+
+    /// Get the position of the second point on the line segment
+    pub fn point2(&self) -> Vec2 {
+        *self.direction * self.half_length
+    }
+}
+
+/// A series of connected line segments in 2D space.
+///
+/// For a version without generics: [`BoxedPolyline2d`]
+#[derive(Clone, Debug)]
+pub struct Polyline2d<const N: usize> {
+    /// The vertices of the polyline
+    pub vertices: [Vec2; N],
+}
+impl<const N: usize> Primitive2d for Polyline2d<N> {}
+
+/// A series of connected line segments in 2D space.
+///
+/// For a version without alloc: [`Polyline2d`]
+#[derive(Clone, Debug)]
+pub struct BoxedPolyline2d {
+    /// The vertices of the polyline
+    pub vertices: Box<[Vec2]>,
+}
+impl Primitive2d for BoxedPolyline2d {}
+
+/// A triangle in 2D space
+#[derive(Clone, Debug)]
+pub struct Triangle2d {
+    /// The vertices of the triangle
+    pub vertices: [Vec2; 3],
+}
+impl Primitive2d for Triangle2d {}
+
+/// A rectangle primitive
+#[doc(alias = "Quad")]
```

Comment:
It's mostly there for anyone looking for a Quad so they get this as its the closest thing to what they might be looking for, since plenty of engines and programs don't offer you a rectangle but just a quad but then only allow you to set width/height anyway.